### PR TITLE
Update log4j version

### PR DESCRIPTION
**Pull request by srnagar**
Updating log4j version to 2.15.0.

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

**Review by JonathanGiles — APPROVED** [review]

**Review by joshfree — APPROVED** [review]
LGTM.  Thank you.

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -75,8 +75,8 @@ org.apache.avro:avro-maven-plugin;1.10.1
 org.apache.commons:commons-compress;1.21
 org.apache.commons:commons-lang3;3.12.0
 org.apache.httpcomponents:httpclient;4.5.13
-org.apache.logging.log4j:log4j-api;2.14.1
-org.apache.logging.log4j:log4j-core;2.14.1
+org.apache.logging.log4j:log4j-api;2.15.0
+org.apache.logging.log4j:log4j-core;2.15.0
 org.apache.logging.log4j:log4j-slf4j-impl;2.14.1
```
Comment (alzimmermsft) [inline eng/versioning/external_dependencies.txt:80]: Does this need to  be updated as well

Comment (srnagar) [reply eng/versioning/external_dependencies.txt]: updated

**Review by JamesBirdsall — APPROVED** [review]

**Review by kasobol-msft — APPROVED** [review]

**Review by moderakh — APPROVED** [review]
Thanks for the quick turn around time for addressing the security vulnerability.

**Review by kushagraThapar — APPROVED** [review]
LGTM

**Comment by kushagraThapar** [comment]
Cosmos test failures are intermittent, we are good to merge this from cosmos side.

**Comment by chenrujun** [comment]
I have a question: 

For these modules included in this PR, can we use `SLF4J` instead of `Log4J`? 
IMU, It's better to not bind to specific logging framework for our libraries.

**Pull request merged**